packages/application-components/src/experimental/components/product-picker-input/product-picker-input.spec.tsx

```diff
@@ -0,0 +1,216 @@
+import type { TRenderAppOptions } from '@commercetools-frontend/application-shell/test-utils';
+
+// import { graphql } from 'msw';
+// import { setupServer } from 'msw/node';
+import React from 'react';
+import {
+  renderApp,
+  fireEvent,
+  screen,
+} from '@commercetools-frontend/application-shell/test-utils';
+import { GRAPHQL_TARGETS } from '@commercetools-frontend/constants';
+import ProductPickerInput from './product-picker-input';
+import FetchProductByPrefix from './fetch-product-by-prefix.ctp.graphql';
+import PrefetchProduct from './prefetch-selected-product.ctp.graphql';
+
+const productId = '6f69c022-b894-4deb-b19f-6335547cb5d7';
+const inputName = 'product-picker-input';
+
+const createTestProps = (custom = {}) => ({
+  name: inputName,
+  value: productId,
+  hasError: false,
+  isReadOnly: false,
+  isClearable: false,
+  isDisabled: false,
+  onChange: jest.fn().mockName('onChange'),
+  onInitialLoadError: jest.fn().mockName('onInitialLoadError'),
+  onLoadError: jest.fn().mockName('onLoadError'),
+  ...custom,
+});
+
+const createTestProduct = (custom = {}) => ({
+  id: productId,
+  key: 'milk-product',
+  masterData: {
+    current: {
+      nameAllLocales: [
+        {
+          locale: 'en',
+          value: 'Milk',
+        },
+      ],
+    },
+  },
+  ...custom,
+});
+
+const createPrefetchedProductQueryMock = (custom = {}) => ({
+  request: {
+    query: PrefetchProduct,
+    context: { target: GRAPHQL_TARGETS.COMMERCETOOLS_PLATFORM },
+    variables: { id: productId },
+  },
+  result: {
+    data: {
+      product: createTestProduct(),
+    },
+  },
+  ...custom,
+});
+
+const createPrefixedSearchProductQueryMock = (custom = {}) => ({
+  request: {
+    query: FetchProductByPrefix,
+    variables: {
+      where:
+        'masterData(current(name(en >= "H" and en < "I"))) or key >= "H" and key < "I"',
+    },
+  },
+  result: {
+    data: {
+      products: {
+        results: [createTestProduct()],
+      },
+    },
+  },
+  ...custom,
+});
+
+const render = (
+  ui: React.ReactElement,
+  renderOptions: Partial<TRenderAppOptions<{}>>
+) => {
+  return renderApp(
+    <>
+      <label htmlFor={inputName}>Product picker</label>
+      {ui}
+    </>,
+    {
+      dataLocale: 'en',
+      ...renderOptions,
+    }
+  );
+};
+
+describe('initial load', () => {
+  describe('when `value` is valid', () => {
+    it('should prefetch product', async () => {
+      const props = createTestProps();
+      render(<ProductPickerInput {...props} />, {
+        mocks: [createPrefetchedProductQueryMock()],
+      });
+      await screen.findByText(/milk/i);
+    });
+  });
+  describe('when `value` is not valid', () => {
+    it('should not prefetch product', async () => {
```

Comment:
Do we also expect to show an error message/state here?

Reply:
with the status quo, I would not recommend that to display to the MC user.
The cases where the value is invalid;

1. user pastes in a invalid ID => do not send a request. displaying an error wouldn't make sense and if anything the consumer should decide that through `onInputChange`
2. the initial value was `null` <= this is a side effect of transformations